metrics/metrics.go

```diff
@@ -0,0 +1,53 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var Enabled = false
+
+func MustRegister(collectors ...prometheus.Collector) {
+	if Enabled {
+		prometheus.MustRegister(collectors...)
+	}
+}
+
+type Metrics struct {
+	listener net.Listener
+}
+
+func New(listener net.Listener) *Metrics {
+	return &Metrics{
+		listener: listener,
+	}
+}
+
+func (m *Metrics) Run(ctx context.Context) error {
+	errCh := make(chan error)
+
+	srv := &http.Server{
+		Addr:    m.listener.Addr().String(),
+		Handler: promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{Registry: prometheus.DefaultRegisterer}),
+		// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
```

Comment:
This comment is misleading as setting `ReadTimeout` doesn't set  `ReadHeaderTimeout` and `IdleTimeout`. Instead `ReadTimeout` is used if `ReadHeaderTimeout` and `IdleTimeout` are not set. IMO, it would be best to remove it.

Reply:
changed